test/lib/jdk/test/lib/process/ProcessTools.java

```diff
@@ -217,14 +217,20 @@ protected void processLine(String line) {
         try {
             if (timeout > -1) {
                 if (timeout == 0) {
-                    latch.await();
+                    while (!latch.await(1, TimeUnit.SECONDS)) {
+                        if (!p.isAlive()) {
+                            //Fail if process finished before printed expected string
+                            latch.countDown();
+                            throw new RuntimeException("Started process is not alive.");
+                        }
+                    }
                 } else {
                     if (!latch.await(Utils.adjustTimeout(timeout), unit)) {
```

Comment:
Checking for liveness in both timeout ==0 and > 0 would be useful.

Reply:
fixed